redis/asyncio/connection.py

```diff
@@ -763,7 +763,11 @@ async def send_packed_command(
             raise ConnectionError(
                 f"Error {err_no} while writing to socket. {errmsg}."
             ) from e
-        except Exception:
+        except BaseException:
+            # The send_packed_command api does not support re-trying a partially
+            # sent message, so there is no point in keeping the connection open.
```

Comment:
Consider: The lack of support is really RESP's (the wire protocol).

Reply:
Actually no.  You can keep the buffer of data (the message) around in a send buffer.  the actual socket.send() message lets you know how much of this has been successfully delivered to underlying layers.  So a re-triable API could, for example, accept a `None` as the data, and then continue sending from where it left off, if it has a buffer in place.  Consider this api:
```python
def send_packed_command(message: Optional[str]):
   """ Send the packed command, or, if None, a previous call must not have completed and the function will
        will resume sending that data over the wire.
  """
  ...
```

Reply:
Yes, on the Python side (e.g. in `Connection`) we could choose to implement a write buffer and "continue where we left off". As [I mentioned](https://github.com/redis/redis-py/issues/2499#issuecomment-1346737191) before, the problem with a write buffer would be the potential of flushing out a command that's stale from an app code's perspective.

What I'd rather see, is being able to, instead of flushing a command that can be stale, to send a short control sequence that would cancel a command that's half-written into the socket, turning it into a no-op. This is the feature that RESP protocol doesn't support.

So:
1. Write buffer: Not an inherent limitation in Python, we simply don't want to do it because it's bad UX.
2. Cancel a half-written command: Inherent limitation of RESP.

Hence the "problem" is RESP, not `send_packed_command`.

Reply:
I see what you are saying, but I'm not even thinking about the "cancel" of the command.  That's something I've never seen and in my mind completely unnecessary.  For that, you can simply disconnect.   I'm primarily thinking about Timeouts, where the transmitting of the command over the wire times out, and you want to maybe do some housekeeping and then continue trying to send the rest of the command.   

In the context of this PR, do you suggest that I change this comment somehow?  Is it confusing?  I always find that code comments help future contributors.  A similar comment would have probably caused me to do my original PR differently :)

Reply:
Yes, I'm discussing how we phrase the comment :) I think I understand now.

**My perspective:**
I called `send` and caught an exception. Next time I call `send`, it might be a different command, at a different time, on a different thread. I wouldn't want `send_packed_command` to flush my previous command at an arbitrary time in the future. I would settle for flushing a short "cancel sequence" instead, but RESP doesn't support that, so instead I'd like it to disconnect.

From that perspective, it's a RESP limitation.

**Your perspective:**
I called `send` and caught an exception. Now I want to retry immediately, but `send_packed_command` didn't tell me _where_ it stopped nor allows me to resume half-way, so I can't and it's a limitation of `send_packed_command`.

From that perspective, it's a `send_packed_command` limitation.

Reply:
Yes, I understand.  Let me try to formulate the comment better.